### ISSUE TITLE
xauth warning and not registering keys on Linux

### ISSUE THREAD

**Issue by mibli** [boppreh/keyboard#14]
I'm running my program with sudo

Initailly I've gotten such result:
```
FileNotFoundError: [Errno 2] No such file or directory: '/root/.Xauthority'
```

then after creating empty .Xauthority file for root
```
Xlib.xauth: warning, no xauthority details available
```

Callbacks I've added arent executed when I press the keys.
Clearly some description of the issue is missing in the readme.

And here's how I'm using the llibrary:
https://github.com/mibli/i3-pyswitch/blob/feature/daemon/src/i3switch/input.py

**Comment by boppreh** [comment]
Hi mibli

`keyboard` does not use or depend on X in any way (except for the secret mouse part, but it seems you are not using that). Are you sure the error is being caused by this library? Can you post a more complete stack trace?

**Comment by boppreh** [comment]
Just to be sure, I ran the module under strace and grepping for `authority`. No matches even in child processes (like `dumpkeys`, used to map the keyboard layout).

I'm fairly confident your error is not being caused by `add_hotkey`, but I'll let you double check before closing the issue.

**Comment by mibli** [comment]
I've confirmed the error isn't caused by keyboard module, but i3ipc module.
You can close the issue, but I still can't get keyboard to react to add_hotkey, even with a simple program:

``` python
import keyboard
import time
import logging

BINDS = []

BINDS.append(keyboard.add_hotkey("shift+h", logging.info, ['pressed']))

while True:
    time.sleep(1)
```

Am I doing something wrong?
Also forgot to mention I'm using i3wm.

**Comment by boppreh** [comment]
I think your issue is on how you reporting. By default the logging module uses a root Logger object (available at `logging.root`) that prints information to the screen. `logging.info` simply passes the message to this root object. However, by default the root object ignores all logs less important than warnings, so your info messages get dropped.

If you want to see info messages simply set the level of the root logger: `logging.root.setLevel(0)`. Or, if you are using Python3, replace `logging.info` with a simple `print` for a quick and dirty replacement.

For testing hotkeys I suggest using a simple:

```
import keyboard
keyboard.add_hotkey('shift+h', print, ['pressed'])
input()
```

The `input()` blocks until you press enter.

I'm closing this issue, but if you have any bugs or suggestions please don't hesitate to comment here, open another issue or send me an email.

**Comment by mibli** [comment]
On the logging.info I corrected myself used logging.error instead. I prefer logging over print because logging seems to work better for multithreaded applications, while print sometimes would not print anything.
I used keyboard.hook() to check if it was catching anything, and found out to my suprise that the module had no "hook" method. So I uninstalled keyboard from pip and manually added submodule to my repo, and it works. So maybe there was something wrong with the module I installed from the repository.
